### PR TITLE
feature: added validations fields from register view

### PR DESCRIPTION
**Pull request by Djonatanabreu**
In this Feature I've:

- Added fields validations from Register View,
- Added separated files for mockies variables and helpers functions,
- cleaned a little more the code from Register View,
- refactored some codes from DropDown component.

### CODE REVIEW

File: src/components/Input/index.tsx [hunk 1]
```diff
@@ -1,36 +1,34 @@
 import { TextInput, TextInputProps, View } from 'react-native';
 import { Colors } from '../constants/Colors';
-import { height, width } from '../constants/Responsive';
+import { width } from '../constants/Responsive';
 import { Text } from '../Text';
 // import { Icon } from '../Icon';
```
Comment (danielanthonyl) [inline src/components/Input/index.tsx:5]: found a comment!

File: src/components/Input/index.tsx [hunk 2]
```diff
@@ -1,36 +1,34 @@
...
 import { styles } from './styles';
 
 export interface InputProps extends TextInputProps {
   small?: boolean;
-  messege?: string;
-  validationField?: boolean;
+  message?: string;
```
Comment (danielanthonyl) [inline src/components/Input/index.tsx:10]: suggestion: shouldn't it be errorMessage?

File: src/views/Register/_helpers/getFormField.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+import { InitialFormState } from './initialState';
+
+export const getFormField = (
+  form: InitialFormState[],
+  fieldIndex: FormIndexes
+) => {
+  return form[fieldIndex];
+};
+
+// formStateObect || formStateField
```
Comment (danielanthonyl) [inline src/views/Register/_helpers/getFormField.ts:11]: found a comment!

File: src/views/Register/_helpers/getFormField.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { FormIndexes } from '../types';
+import { InitialFormState } from './initialState';
+
+export const getFormField = (
+  form: InitialFormState[],
```
Comment (danielanthonyl) [inline src/views/Register/_helpers/getFormField.ts:5]: suggestion: type InitialFormState could be renamed to `FormField`

File: src/views/Register/index.tsx [hunk 5]
```diff
@@ -136,12 +69,17 @@ export const Register = () => {
     console.log('Privacy Policy');
   };
 
-  const onBackHandler = () => {
-    console.log('Back');
-  };
+  const onReceiveNews = () =>
+    // send to API
+    console.log('Receive News');
```
Comment (danielanthonyl) [inline src/views/Register/index.tsx:74]: suggestion: no `console.logs`

File: src/views/Register/index.tsx [hunk 6]
```diff
@@ -136,12 +69,17 @@ export const Register = () => {
     console.log('Privacy Policy');
   };
 
-  const onBackHandler = () => {
-    console.log('Back');
-  };
+  const onReceiveNews = () =>
+    // send to API
```
Comment (danielanthonyl) [inline src/views/Register/index.tsx:73]: suggestion: add `TODO:`

File: src/views/Register/index.tsx [hunk 7]
```diff
@@ -136,12 +69,17 @@ export const Register = () => {
...
+    console.log('Receive News');
+
+  const enabled = () => {
```
Comment (danielanthonyl) [inline src/views/Register/index.tsx:76]: suggestion: change `enabled` to something more descriptive like `enableConfirmButton`

File: src/views/Register/index.tsx [hunk 8]
```diff
@@ -11,35 +11,17 @@ import { Input } from '../../components/Input';
...
   const [form, setForm] = useState(initialFormState);
-  const [errorMessage, setErrorMessage] = useState('');
-  const mockCountries = [
-    'Brazil',
-    'French',
-    'Italy',
-    'Australia',
-    'United State',
-  ];
+  const [checked, setChecked] = useState(false);
```
Comment (danielanthonyl) [inline src/views/Register/index.tsx:24]: suggestion: checked what?

File: src/components/Input/index.tsx (hunk 1 above)
Comment (Djonatanabreu) [reply src/components/Input/index.tsx:5]: fixed at 
[e0786b6](https://github.com/AbreuLetitia/legendary-games/pull/9/commits/e0786b6c97255758c78f8d535e010a8774944d00)

File: src/components/Input/index.tsx (hunk 2 above)
Comment (Djonatanabreu) [reply src/components/Input/index.tsx:10]: fixed at
[e0786b6](https://github.com/AbreuLetitia/legendary-games/pull/9/commits/e0786b6c97255758c78f8d535e010a8774944d00)

File: src/views/Register/_helpers/getFormField.ts (hunk 3 above)
Comment (Djonatanabreu) [reply src/views/Register/_helpers/getFormField.ts:11]: fixed at
[e0786b6](https://github.com/AbreuLetitia/legendary-games/pull/9/commits/e0786b6c97255758c78f8d535e010a8774944d00)

File: src/views/Register/_helpers/getFormField.ts (hunk 4 above)
Comment (Djonatanabreu) [reply src/views/Register/_helpers/getFormField.ts:5]: fixed at
[e0786b6](https://github.com/AbreuLetitia/legendary-games/pull/9/commits/e0786b6c97255758c78f8d535e010a8774944d00)

File: src/views/Register/index.tsx (hunk 5 above)
Comment (Djonatanabreu) [reply src/views/Register/index.tsx:74]: fixed at
[e0786b6](https://github.com/AbreuLetitia/legendary-games/pull/9/commits/e0786b6c97255758c78f8d535e010a8774944d00)

File: src/views/Register/index.tsx (hunk 6 above)
Comment (Djonatanabreu) [reply src/views/Register/index.tsx:73]: fixed at
[e0786b6](https://github.com/AbreuLetitia/legendary-games/pull/9/commits/e0786b6c97255758c78f8d535e010a8774944d00)

File: src/views/Register/index.tsx (hunk 7 above)
Comment (Djonatanabreu) [reply src/views/Register/index.tsx:76]: fixed at
[e0786b6](https://github.com/AbreuLetitia/legendary-games/pull/9/commits/e0786b6c97255758c78f8d535e010a8774944d00)

File: src/views/Register/index.tsx (hunk 8 above)
Comment (Djonatanabreu) [reply src/views/Register/index.tsx:24]: fixed at
[e0786b6](https://github.com/AbreuLetitia/legendary-games/pull/9/commits/e0786b6c97255758c78f8d535e010a8774944d00)

**Review by danielanthonyl — APPROVED** [review]

**Pull request merged**